### PR TITLE
Adds Component for Submodel Service

### PR DESCRIPTION
**Pull request by FriedJannik**

### CODE REVIEW

File: basyx.submodelservice/basyx.submodelservice-http/pom.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	</parent>
+
+	<artifactId>basyx.submodelservice-http</artifactId>
+
+	<dependencies>
+		<dependency>
+			<groupId>org.eclipse.digitaltwin.basyx</groupId>
+			<artifactId>basyx.http</artifactId>
+		</dependency>
+		<dependency>
```
Comment (jannisjung) [inline basyx.submodelservice/basyx.submodelservice-http/pom.xml:19]: unused; remove

File: basyx.submodelservice/basyx.submodelservice-http/pom.xml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+			<groupId>org.eclipse.digitaltwin.basyx</groupId>
+			<artifactId>basyx.http</artifactId>
+			<scope>test</scope>
+			<classifier>tests</classifier>
+		</dependency>
+		<dependency>
```
Comment (jannisjung) [inline basyx.submodelservice/basyx.submodelservice-http/pom.xml:25]: declare as used

File: basyx.submodelservice/basyx.submodelservice-http/pom.xml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+			<groupId>org.eclipse.digitaltwin.basyx</groupId>
+			<artifactId>basyx.submodelservice-core</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.eclipse.digitaltwin.basyx</groupId>
+			<artifactId>basyx.submodelservice-core</artifactId>
+			<classifier>tests</classifier>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
```
Comment (jannisjung) [inline basyx.submodelservice/basyx.submodelservice-http/pom.xml:35]: unused; remove

File: basyx.submodelservice/basyx.submodelservice-http/pom.xml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+			<groupId>org.eclipse.digitaltwin.basyx</groupId>
+			<artifactId>basyx.submodelservice-backend-inmemory</artifactId>
+			<scope>test</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-context</artifactId>
+		</dependency>
+		<dependency>
```
Comment (jannisjung) [inline basyx.submodelservice/basyx.submodelservice-http/pom.xml:45]: unused; remove

File: basyx.submodelservice/basyx.submodelservice-http/pom.xml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+		</dependency>
+		<dependency>
```
Comment (jannisjung) [inline basyx.submodelservice/basyx.submodelservice-http/pom.xml:49]: unused; remove

File: basyx.submodelservice/basyx.submodelservice-http/pom.xml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+			<groupId>org.springdoc</groupId>
+			<artifactId>springdoc-openapi-ui</artifactId>
+		</dependency>
+		<dependency>
```
Comment (jannisjung) [inline basyx.submodelservice/basyx.submodelservice-http/pom.xml:53]: unused; remove

File: basyx.submodelservice/basyx.submodelservice-http/pom.xml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+			<groupId>javax.validation</groupId>
+			<artifactId>validation-api</artifactId>
+		</dependency>
+		<dependency>
```
Comment (jannisjung) [inline basyx.submodelservice/basyx.submodelservice-http/pom.xml:57]: unused; remove

File: basyx.submodelservice/basyx.submodelservice-http/pom.xml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+			<groupId>org.eclipse.digitaltwin.aas4j</groupId>
+			<artifactId>dataformat-json</artifactId>
+		</dependency>
+		<dependency>
```
Comment (jannisjung) [inline basyx.submodelservice/basyx.submodelservice-http/pom.xml:61]: unused; remove

File: basyx.submodelservice/basyx.submodelservice-http/pom.xml [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+			<groupId>org.apache.httpcomponents.client5</groupId>
+			<artifactId>httpclient5</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
```
Comment (jannisjung) [inline basyx.submodelservice/basyx.submodelservice-http/pom.xml:66]: unused; remove

**Review by jannisjung — CHANGES_REQUESTED** [review]
Please Resolve the review Remarks

**Review by jannisjung — APPROVED** [review]
approved

File: basyx.submodelservice/basyx.submodelservice-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/http/SubmodelServiceSubmodelElementsTestSuiteHTTP.java [hunk 10]
```diff
@@ -0,0 +1,553 @@
...
+		CloseableHttpResponse deleteResponse = BaSyxHttpTestUtils
+				.executeDeleteOnURL(createSpecificSubmodelElementURL(nestedIdShortPathInCollection));
+		assertEquals(HttpStatus.NO_CONTENT.value(), deleteResponse.getCode());
+
+		CloseableHttpResponse fetchedNestedInCollectionResponse = requestSubmodelElement(nestedIdShortPathInCollection);
+		assertEquals(HttpStatus.NOT_FOUND.value(), fetchedNestedInCollectionResponse.getCode());
+	}
+
+	@Test
+	public void delteNestedSubmodelElementFromSubmodelElementList()
```
Comment (FrankSchnicke) [inline basyx.submodelservice/basyx.submodelservice-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/http/SubmodelServiceSubmodelElementsTestSuiteHTTP.java]: Typo in "delete"

File: basyx.submodelservice/basyx.submodelservice-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/http/SubmodelServiceSubmodelElementsTestSuiteHTTP.java [hunk 11]
```diff
@@ -0,0 +1,553 @@
...
+			throws IOException {
+		// delete first element of the submodel element list
```
Comment (FrankSchnicke) [inline basyx.submodelservice/basyx.submodelservice-http/src/test/java/org/eclipse/digitaltwin/basyx/submodelservice/http/SubmodelServiceSubmodelElementsTestSuiteHTTP.java]: Please remove this comment

**Pull request merged**